internal/testrunner/runners/system/runner.go

```diff
@@ -933,14 +933,16 @@ func (r *runner) prepareScenario(ctx context.Context, config *testConfig, svcInf
 	agent := agents[0]
 	logger.Debugf("Selected enrolled agent %q", agent.ID)
 
-	r.unenrollAgentHandler = func(ctx context.Context) error {
-		if !r.options.RunIndependentElasticAgent {
+	r.removeAgentHandler = func(ctx context.Context) error {
+		// When not using independent agents, service deployers like kubernetes or custom agents create new Elastic Agent
+		createdNewAgent := svcInfo.Agent.Host.NamePrefix == "docker-custom-agent" || svcInfo.Agent.Host.NamePrefix == "kind-control-plane"
```

Comment:
Nit. Depending on the name of the agent may be fragile, we may forget to modify this if the set of names change.

Reply:
Or maybe change it to be `Independent`

`svcInfo.Agent.Independent` (true or false)

Reply:
Perfect, I'll merge this PR as it is then, and I'll create a follow-up PR to remove those agent names with one of these options and run the tests with integrations again.

Reply:
Created PR #1776